### PR TITLE
feature/CVSB-7521

### PR DESCRIPTION
**Pull request by DeepikaSinghD**
Automation Test Cases for CVSB-7521

### CODE REVIEW

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 1]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+        testResultsSteps.statusCodeShouldBe(200);
+
+        // Verify LEC test field values match the expected values.
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeId", "39");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeName", "Low Emissions Certificate (LEC) with annual test");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testCode", "lbp");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].certificateNumber", testResultsSteps.getTestNumber());
+    }
+
+    @WithTag("In_Test")
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: please remove this line once all tests passed for the feature

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 2]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+    @Title("CVSB-7521 - TC - AC1 - Generate certificate (PASS) - LEC with linked test - HGV")
+    @Test
+    public void testResults_LEC_Linked_Test_HGV_Pass_Certificate_Number() {
+
+        // Read the base test result JSON.
+        String testResultRecord = GenericData.readJsonValueFromFile("test-results_LEC_linked_test_hgv_pass_7521.json", "$");
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: please commit the .json file

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 3]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+    }
+
+
+    @WithTag("In_Test")
+    @Title("CVSB-7521 - TC - AC1 - Generate certificate (PASS) - LEC not linked test - HGV")
+    @Test
+    public void testResults_LEC_not_Linked_Test_HGV_Pass_Certificate_Number() {
+
+        // Read the base test result JSON.
+        String testResultRecord = GenericData.readJsonValueFromFile("test-results_LEC_not_linked_test_hgv_pass_7521.json", "$");
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: please commit the .json file

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 4]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+
+    }
+
+    @WithTag("In_Test")
+    @Title("CVSB-7521 - TC - AC1 - Generate certificate (PASS) - linked test - PSV")
+    @Test
+    public void testResults_LEC_Linked_Test_PSV_Pass_Certificate(){
+
+        // Read the base test result JSON.
+        String testResultRecord = GenericData.readJsonValueFromFile("test-results_LEC_linked_psv_pass_7521.json", "$");
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: please commit the .json file

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 5]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+    }
+
+    @WithTag("In_Test")
+    @Title("CVSB-7521 - TC - AC1 - Generate certificate (PASS) - not linked test - PSV")
+    @Test
+
+    public void testResults_LEC_not_Linked_Test_PSV_Pass(){
+
+        // Read the base test result JSON.
+        String testResultRecord = GenericData.readJsonValueFromFile("test-results_LEC_not_linked_psv_paas_7521.json", "$");
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: please commit the .json file

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 6]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+    }
+
+    @WithTag("In_Test")
+    @Title("CVSB-7521 - TC - AC12 - Generate certificate (FAIL) - linked test - HGV")
+    @Test
+
+    public void testResults_LEC_Linked_Test_HGV_Fail_Certificate_Number() {
+
+        // Read the base test result JSON.
+        String testResultRecord = GenericData.readJsonValueFromFile("test-results_LEC_linked_test_hgv_fail_7521.json", "$");
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: please commit the .json file

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 7]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].certificateNumber", testResultsSteps.getTestNumber());
+    }
+
+    @WithTag("In_Test")
+    @Title("CVSB-7521 - TC - AC1 - Generate certificate (FAIL) - LEC not linked test - HGV")
+    @Test
+    public void testResults_LEC_not_Linked_Test_HGV_Fail_Certificate_Num() {
+
+        // Read the base test result JSON.
+        String testResultRecord = GenericData.readJsonValueFromFile("test-results_LEC_not_linked_hgv_fail_7521.json", "$");
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: please commit the .json file

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 8]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+
+    }
+
+    @WithTag("In_Test")
+    @Title("CVSB-7521 - TC - AC1 - Generate certificate (FAIL) - linked test - PSV")
+    @Test
+    public void testResults_LEC_Linked_Test_PSV_Fail_Certificate_Number(){
+
+        // Read the base test result JSON.
+    String testResultRecord = GenericData.readJsonValueFromFile("test-results_LEC_linked_psv_fail_7521.json", "$");
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: please commit the .json file

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 9]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+
+    @WithTag("In_Test")
+    @Title("CVSB-7521 - TC - AC1 - Generate certificate (FAIL) - not linked test - PSV")
+    @Test
+
+    public void testResults_LEC_not_Linked_Test_PSV_Fail_Certificate_Number(){
+
+
+        // Read the base test result JSON.
+        String testResultRecord = GenericData.readJsonValueFromFile("test-results_LEC_not_linked_psv_fail_7521.json", "$");
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: please commit the .json file

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 10]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
         testResultsSteps.statusCodeShouldBe(400);
         testResultsSteps.validateErrorText("Fuel Type not present on LEC test type");
     }
+
+    @WithTag("In_Test")
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: remove this line once all the tests for the feature are passing in CI

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 11]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+
+    // Verify LEC test field values match the expected values.
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeId", "39");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeName", "Low Emissions Certificate (LEC) with annual test");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testCode", "lcp");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].certificateNumber", testResultsSteps.getTestNumber());
+
+}
+
+    @WithTag("In_Test")
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: remove this line once all the tests for the feature are passing in CI

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 12]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeId", "45");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeName", "Low Emissions Certificate (LEC)");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testCode", "lev");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].certificateNumber", testResultsSteps.getTestNumber());
+
+
+
+    }
+
+    @WithTag("In_Test")
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: remove this line once all the tests for the feature are passing in CI

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 13]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+        testResultsSteps.statusCodeShouldBe(200);
+
+        // Verify LEC test field values match the expected values.
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeId", "44");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeName", "Low Emissions Certificate (LEC) with annual test");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testCode", "ldv");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].certificateNumber", testResultsSteps.getTestNumber());
+    }
+
+    @WithTag("In_Test")
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: remove this line once all the tests for the feature are passing in CI

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 14]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+
+        // Verify LEC test field values match the expected values.
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeId", "39");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeName", "Low Emissions Certificate (LEC) with annual test");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testCode", "lbp");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].certificateNumber", testResultsSteps.getTestNumber());
+
+    }
+
+    @WithTag("In_Test")
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: remove this line once all the tests for the feature are passing in CI

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 15]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeId", "39");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeName", "Low Emissions Certificate (LEC) with annual test");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testCode", "lcp");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].certificateNumber", testResultsSteps.getTestNumber());
+
+
+
+    }
+
+    @WithTag("In_Test")
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: remove this line once all the tests for the feature are passing in CI

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 16]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+        // Verify LEC test field values match the expected values.
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeId", "45");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeName", "Low Emissions Certificate (LEC)");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testCode", "lev");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].certificateNumber", testResultsSteps.getTestNumber());
+
+
+    }
+
+    @WithTag("In_Test")
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: remove this line once all the tests for the feature are passing in CI

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 17]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+
+        // Verify LEC test field values match the expected values.
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeId", "44");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeName", "Low Emissions Certificate (LEC) with annual test");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testCode", "ldv");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].certificateNumber", testResultsSteps.getTestNumber());
+    }
+
+
+    @WithTag("In_Test")
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: remove this line once all the tests for the feature are passing in CI

File: src/test/java/testresults/PostTestResultsLEC.java [hunk 18]
```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
...
+        testResultsSteps.statusCodeShouldBe(200);
+
+        // Verify LEC test field values match the expected values.
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeId", "45");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeName", "Low Emissions Certificate (LEC)");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testCode", "lev");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].certificateNumber", testResultsSteps.getTestNumber());
+    }
+
+    @WithTag("In_Test")
```
Comment (bfloreaDVSA) [inline src/test/java/testresults/PostTestResultsLEC.java]: remove this line once all the tests for the feature are passing in CI

File: src/test/java/testresults/PostTestResultsLEC.java (hunk 2 above)
Comment (DeepikaSinghD) [reply src/test/java/testresults/PostTestResultsLEC.java]: commit done

File: src/test/java/testresults/PostTestResultsLEC.java (hunk 3 above)
Comment (DeepikaSinghD) [reply src/test/java/testresults/PostTestResultsLEC.java]: commit done

File: src/test/java/testresults/PostTestResultsLEC.java (hunk 4 above)
Comment (DeepikaSinghD) [reply src/test/java/testresults/PostTestResultsLEC.java]: commit done

File: src/test/java/testresults/PostTestResultsLEC.java (hunk 5 above)
Comment (DeepikaSinghD) [reply src/test/java/testresults/PostTestResultsLEC.java]: commit done

File: src/test/java/testresults/PostTestResultsLEC.java (hunk 6 above)
Comment (DeepikaSinghD) [reply src/test/java/testresults/PostTestResultsLEC.java]: commit done

File: src/test/java/testresults/PostTestResultsLEC.java (hunk 7 above)
Comment (DeepikaSinghD) [reply src/test/java/testresults/PostTestResultsLEC.java]: commit done

File: src/test/java/testresults/PostTestResultsLEC.java (hunk 1 above)
Comment (DeepikaSinghD) [reply src/test/java/testresults/PostTestResultsLEC.java]: Changes done.

File: src/test/java/testresults/PostTestResultsLEC.java (hunk 14 above)
Comment (DeepikaSinghD) [reply src/test/java/testresults/PostTestResultsLEC.java]: Changes done.

Comment (DeepikaSinghD) [reply src/test/java/testresults/PostTestResultsLEC.java]: Test cases executed in feature, please refer - https://jenkins.cvs.dvsacloud.uk/job/UPDATE__BRANCH/job/job_feature_test_backend/277/

In_Test tag removed.

File: src/test/java/testresults/PostTestResultsLEC.java (hunk 15 above)
Comment (DeepikaSinghD) [reply src/test/java/testresults/PostTestResultsLEC.java]: Test cases executed in feature, please refer - https://jenkins.cvs.dvsacloud.uk/job/UPDATE__BRANCH/job/job_feature_test_backend/277/

In_Test tag removed.

File: src/test/java/testresults/PostTestResultsLEC.java (hunk 16 above)
Comment (DeepikaSinghD) [reply src/test/java/testresults/PostTestResultsLEC.java]: Test cases executed in feature, please refer - https://jenkins.cvs.dvsacloud.uk/job/UPDATE__BRANCH/job/job_feature_test_backend/277/

In_Test tag removed.

File: src/test/java/testresults/PostTestResultsLEC.java (hunk 17 above)
Comment (DeepikaSinghD) [reply src/test/java/testresults/PostTestResultsLEC.java]: Test cases executed in feature, please refer - https://jenkins.cvs.dvsacloud.uk/job/UPDATE__BRANCH/job/job_feature_test_backend/277/

In_Test tag removed.

File: src/test/java/testresults/PostTestResultsLEC.java (hunk 18 above)
Comment (DeepikaSinghD) [reply src/test/java/testresults/PostTestResultsLEC.java]: Test cases executed in feature, please refer - https://jenkins.cvs.dvsacloud.uk/job/UPDATE__BRANCH/job/job_feature_test_backend/277/

In_Test tag removed.

File: src/test/java/testnumber/PostTestNumber.java [hunk 19]
```diff
@@ -385,6 +386,7 @@ public void validTestNumberGeneratedForAnnualWithCertificatePRSIsEqualToCertific
         testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].certificateNumber", testResultsSteps.getTestNumber());
     }
 
+    @Ignore("Deprecated by CVSB-731")
```
Comment (dpanzaru-deloitte) [inline src/test/java/testnumber/PostTestNumber.java:389]: If you put @Ignore here please remove @Test so it won't appear in the test report.

File: src/main/resources/loader/develop/test-results_LEC_linked_psv_fail_7521.json [hunk 20]
```diff
@@ -0,0 +1,116 @@
...
+            }
+          },
+          "stdForProhibition":false,
+          "prs":false,
+          "prohibitionIssued":false
+        }
+      ]
+    }
+  ]
+}
```
Comment (dpanzaru-deloitte) [inline src/main/resources/loader/develop/test-results_LEC_linked_psv_fail_7521.json:116]: Add empty new line at the end of all the newly added payload files

**Pull request merged**